src/comm/Connection.cc

```diff
@@ -133,9 +127,8 @@ Comm::Connection::setPeer(CachePeer *p)
     if (getPeer() == p)
         return;
 
-    cbdataReferenceDone(peer_);
     if (p) {
-        peer_ = cbdataReference(p);
+        peer_ = p;
```

Comment:
This should be the only line left in this method AFAICT. Let RefCount deal with assignment to self and such. I would add a trailing `// may stay unchanged or become nil` comment.

Adding debugging might be useful but is outside this PR scope.

Reply:
Ok.